tools/SampleAnalyzer/Commons/DataFormat/MCEventFormat.h

```diff
@@ -37,206 +35,206 @@
-    multiweights_.Reset();
-    MET_.Reset();
-    MHT_.Reset();
-    TET_  = 0.;
-    THT_  = 0.;
-    Meff_ = 0.;
-  }
+        /// List of weights
+        WeightCollection multiweights_;
 
-  /// Displaying data member values
-  void Print() const
-  {
-    INFO << "nparts="       << particles_.size()
-         << " - processId=" << processId_
-         << " - weight="    << weight_
-         << " - scale="     << scale_
-         << " - alphaQED="  << alphaQED_
-         << " - alphaQCD="  << alphaQCD_ << endmsg;
-    INFO << "nweights=" << multiweights_.size() << endmsg;
-  }
+        // -------------------------------------------------------------
+        //                      method members
+        // -------------------------------------------------------------
+    public:
+        /// Constructor withtout arguments
+        MCEventFormat()
+        {
+            processId_ = 0;
+            // weight_ = 1.;
+            scale_ = 0.;
+            alphaQED_ = 0.;
+            alphaQCD_ = 0.;
+            TET_ = 0.;
+            THT_ = 0.;
+            Meff_ = 0.;
+        }
+
+        /// Destructor
+        ~MCEventFormat() {}
+
+        /// Accessor to the Missing Transverse Energy (read-only)
+        const MCParticleFormat &MET() const { return MET_; }
+
+        /// Accessor to the Missing Hadronic Transverse Energy (read-only)
+        const MCParticleFormat &MHT() const { return MHT_; }
+
+        /// Accessor to the Total Transverse Energy (read-only)
+        const MAfloat64 &TET() const { return TET_; }
+
+        /// Accessor to the Total Hadronic Transverse Energy (read-only)
+        const MAfloat64 &THT() const { return THT_; }
+
+        /// Accessor to the Total effective mass (read-only)
+        const MAfloat64 &Meff() const { return Meff_; }
+
+        /// Accessor to the Missing Transverse Energy
+        MCParticleFormat &MET() { return MET_; }
+
+        /// Accessor to the Missing Hadronic Transverse Energy
+        MCParticleFormat &MHT() { return MHT_; }
+
+        /// Accessor to the Total Transverse Energy
+        MAfloat64 &TET() { return TET_; }
+
+        /// Accessor to the Total Hadronic Transverse Energy
+        MAfloat64 &THT() { return THT_; }
+
+        /// Accessor to the Total effective mass
+        MAfloat64 &Meff() { return Meff_; }
+
+        /// Accessor to the process identity
+        const MAuint32 &processId() const { return processId_; }
+
+        /// Accessor to the event weight
+        const MAfloat64 &weight() const { return multiweights_.Get(0); }
+
+        /// Accessor to the scale
+        const MAfloat64 &scale() const { return scale_; }
+
+        /// Accessor to alpha_QED
+        const MAfloat64 &alphaQED() const { return alphaQED_; }
+
+        /// Accessor to alpha_QCD
+        const MAfloat64 &alphaQCD() const { return alphaQCD_; }
+
+        /// Accessor to multiweights
+        const WeightCollection &multiweights() const { return multiweights_; }
```

Comment:
Why no method ``` WeightCollection& multiweights() {return multiweights_; }```? We have the `weights()` equivalent...

Reply:
In fact I don't understand why we have so many functions doing the same thing.

Reply:
Mostly because they were there before and I didnt see them :D